benchmarks/expected-plans/q12.txt

```diff
@@ -7,7 +7,7 @@
 |               |       Projection: lineitem.l_shipmode, orders.o_orderpriority                                                                                                                                                                                                                                                                                                               |
 |               |         Inner Join: lineitem.l_orderkey = orders.o_orderkey                                                                                                                                                                                                                                                                                                                 |
 |               |           Projection: lineitem.l_orderkey, lineitem.l_shipmode                                                                                                                                                                                                                                                                                                              |
-|               |             Filter: (lineitem.l_shipmode = Utf8("SHIP") OR lineitem.l_shipmode = Utf8("MAIL")) AND lineitem.l_commitdate < lineitem.l_receiptdate AND lineitem.l_shipdate < lineitem.l_commitdate AND lineitem.l_receiptdate >= Date32("8766") AND lineitem.l_receiptdate < Date32("9131")                                                                                  |
+|               |             Filter: (lineitem.l_shipmode = Utf8("MAIL") OR lineitem.l_shipmode = Utf8("SHIP")) AND lineitem.l_commitdate < lineitem.l_receiptdate AND lineitem.l_shipdate < lineitem.l_commitdate AND lineitem.l_receiptdate >= Date32("8766") AND lineitem.l_receiptdate < Date32("9131")                                                                                  |
```

Comment:
It will be better if we can keep original order like `EliminateDuplicatedExpr`

Reply:
I think to be able to do that we need to do merging and expanding of OR / LIST statements in one pass. This seems rather complicated to me at the moment. I'll check`EliminateDuplicatedExpr` for some inspiration.

Reply:
Also, pretty sure that the current code actually preserves the ordering. Previous code messes up the ordering of IN LIST. Actual query:

https://github.com/apache/arrow-datafusion/blob/main/benchmarks/queries/q12.sql
```
select
    l_shipmode,
    sum(case
            when o_orderpriority = '1-URGENT'
                or o_orderpriority = '2-HIGH'
                then 1
            else 0
        end) as high_line_count,
    sum(case
            when o_orderpriority <> '1-URGENT'
                and o_orderpriority <> '2-HIGH'
                then 1
            else 0
        end) as low_line_count
from
    lineitem
        join
    orders
    on
            l_orderkey = o_orderkey
where
        l_shipmode in ('MAIL', 'SHIP')
  and l_commitdate < l_receiptdate
  and l_shipdate < l_commitdate
  and l_receiptdate >= date '1994-01-01'
  and l_receiptdate < date '1995-01-01'
group by
    l_shipmode
order by
    l_shipmode;
```

Reply:
Thank your explanation.

Reply:
But yeah it won't preserve ordering of complex OR statements like: A OR (B OR C). It will just translates this into A OR B OR C.